pyproject.toml

```
@@ -69,95 +70,148 @@ skip_glob = [
+    "TCH",    # flake8-type-checking
+    "TD",     # flake8-todos
+    "TID",    # flake8-tidy-imports
+    "TRY",    # tryceratops
+    "TRY002", # Disallow raising vanilla Exception. Create or use a custom exception instead.
+    "TRY003", # Disallow vanilla string passing. Prefer kwargs to the exception constructur.
+    "UP",     # pyupgrade
+    "W",      # pycodestyle (warnings)
+    "YTT",    # flake8-2020
 ]
 
-[tool.ruff.per-file-ignores]
-"__init__.py" = [
-    "F401", # Permit unused imports in `__init__.py` files
+ignore = [
+    # For rules reference, see https://docs.astral.sh/ruff/rules/
+
+    # These we don't agree with or don't want to prioritize to enforce:
+    "ANN003",  # kwargs missing type annotations
+    "ANN101",  # Type annotations for 'self' args
+    "ANN102",  # Type annotations for 'cls' args
+    "COM812",  # Because it conflicts with ruff auto-format
+    "EM",      # flake8-errmsgs (may reconsider later)
+    "DJ",      # Django linting
+    "G",       # flake8-logging-format
+    "ISC001",  # Conflicts with ruff auto-format
+    "NPY",     # NumPy-specific rules
+    "PIE790",  # Allow unnecssary 'pass' (sometimes useful for readability)
+    "PERF203", # exception handling in loop
+    "S",       # flake8-bandit (noisy, security related)
+    "SIM910",  # Allow "None" as second argument to Dict.get(). "Explicit is better than implicit."
+    "TD002",   # Require author for TODOs
+    "TRIO",    # flake8-trio (opinionated, noisy)
+    "INP001",  # Dir 'examples' is part of an implicit namespace package. Add an __init__.py.
+
+    # TODO: Consider re-enabling these before release:
+    "A003",    # Class attribute 'type' is shadowing a Python builtin
+    "BLE001",  # Do not catch blind exception: Exception
+    "ERA001",  # Remove commented-out code
+    "FIX002",  # Allow "TODO:" until release (then switch to requiring links via TDO003)
+    "PLW0603", # Using the global statement to update _cache is discouraged
+    "TD003",   # Require links for TODOs # TODO: Re-enable when we disable FIX002
+
+    "UP007", # Allow legacy `Union[a, b]` and `Optional[a]` for Pydantic, until we drop Python 3.9 (Pydantic doesn't like it)
+]
+fixable = ["ALL"]
+unfixable = [
+    "ERA001", # Commented-out code (avoid silent loss of code)
+    "T201",   # print() calls (avoid silent loss of code / log messages)
 ]
 
-[tool.ruff.flake8-annotations]
-allow-star-arg-any = true
-mypy-init-return = true
-suppress-dummy-args = true
-
-[tool.ruff.flake8-pytest-style]
-parametrize-values-type = "tuple"
+line-length = 140
+extend-exclude = ["docs", "test", "tests"]
+dummy-variable-rgx = "^(_+|(_+[a-zA-Z0-9_]*[a-zA-Z0-9]+?))$"
 
 [tool.ruff.isort]
-known-first-party = [
-    "airbyte",
-    "airbyte_cdk",
-    "airbyte_protocol",
-    "airbyte_api_client",
-    "connector_ops",
-    "pipelines",
+force-sort-within-sections = false
+lines-after-imports = 2
+known-first-party = ["airbyte_cdk", "airbyte_protocol"]
```

Suggesting add back other first party libraries.

```suggestion
known-first-party = [
    "airbyte",
    "airbyte_cdk",
    "airbyte_protocol",
    "airbyte_api_client",
    "connector_ops",
    "pipelines",
]
```